src/solana-client.ts

```diff
@@ -0,0 +1,532 @@
+    platformAuthority: Keypair
+  ): Promise<string> {
+    const [rewardPoolPDA] = this.getRewardPoolPDA();
+
+    const data = Buffer.alloc(1 + 1);
+    data.writeUint8(RewardPoolInstruction.UpdatePlatformFee, 0);
+    data.writeUint8(newFeePercentage, 1);
+
+    const instruction = new TransactionInstruction({
+      keys: [
+        { pubkey: rewardPoolPDA, isSigner: false, isWritable: true },
+        { pubkey: platformAuthority.publicKey, isSigner: true, isWritable: false },
+      ],
+      programId: this.programId,
+      data,
+    });
+
+    const transaction = new Transaction().add(instruction);
+    transaction.recentBlockhash = (
+      await this.connection.getLatestBlockhash()
+    ).blockhash;
+    transaction.feePayer = platformAuthority.publicKey;
+
+    const signature = await sendAndConfirmTransaction(
+      this.connection,
+      transaction,
+      [platformAuthority]
+    );
+
+    return signature;
+  }
+
+  // Helper method to create reward vault
+  async createRewardVault(
+    tokenMint: PublicKey,
+    platformAuthority: Keypair
+  ): Promise<string> {
+    const [rewardVaultPDA] = this.getRewardVaultPDA(tokenMint);
+
+    const instruction = createAssociatedTokenAccountInstruction(
+      platformAuthority.publicKey,
+      rewardVaultPDA,
+      rewardVaultPDA,
+      tokenMint
+    );
+
+    const transaction = new Transaction().add(instruction);
+    transaction.recentBlockhash = (
+      await this.connection.getLatestBlockhash()
+    ).blockhash;
+    transaction.feePayer = platformAuthority.publicKey;
+
+    const signature = await sendAndConfirmTransaction(
+      this.connection,
+      transaction,
+      [platformAuthority]
+    );
+
+    return signature;
+  }
+
+  // Helper method to get reward vault balance
+  async getRewardVaultBalance(tokenMint: PublicKey): Promise<BN> {
+    const [rewardVaultPDA] = this.getRewardVaultPDA(tokenMint);
+
+    try {
+      const accountInfo = await this.connection.getAccountInfo(rewardVaultPDA);
+      if (!accountInfo) {
+        return new BN(0);
+      }
+
+      const tokenAccount = await getAccount(this.connection, rewardVaultPDA);
+      return new BN(tokenAccount.amount.toString());
+    } catch (error) {
+      console.error('Failed to get reward vault balance:', error);
+      return new BN(0);
+    }
+  }
+}
+
+// Deserialization helpers (these would need to be implemented based on your serialization format)
+// For now, using placeholder implementations
+export namespace RewardPool {
+  export function deserialize(data: Buffer): RewardPool {
+    // Example layout:
+    // 0: isInitialized (u8)
+    // 1-32: platformAuthority (Pubkey)
+    // 33: platformFeePercentage (u8)
+    // 34-41: totalRewardsDistributed (u64, LE)
+    // 42-49: totalPlatformFeesCollected (u64, LE)
+    // 50: isPaused (u8)
+    return {
+      isInitialized: data[0] === 1,
+      platformAuthority: new PublicKey(data.slice(1, 33)),
+      platformFeePercentage: data[33],
```

Comment:
The manual slicing (`data[33]`, `slice(34, 42)`, etc.) is hard to maintain and error-prone. I’d recommend using a layout library to make these structures more declarative and less brittle.